nrich-search/src/main/java/net/croz/nrich/search/repository/JpaSearchExecutor.java

```diff
@@ -42,10 +42,13 @@ public class JpaSearchExecutor<T> implements SearchExecutor<T> {
 
     private final EntityManager entityManager;
 
+    private final Class<T> domainType;
```

Comment:
Maybe use  `domainClass` as a property name here and in `JpaStringSearchExecutor`?